src/nvim/message.c

```diff
@@ -131,9 +131,8 @@ static int msg_grid_scroll_discount = 0;
 
 static void ui_ext_msg_set_pos(int row, bool scrolled)
 {
-  char buf[MAX_MCO];
+  char buf[MAX_MCO] = { 0 };
```

Comment:
What is being fixed here? The old code is clearer IMHO and how char2bytes typically is used.

Reply:
The old code was writing past the end of the buffer when `size` is 6 (`MAX_MCO`).  Since we're using this to construct a `String`, the NUL termination isn't strictly necessary, so I just zero-initialized `buf`.  If you'd rather, I can change the size of `buf` to `MAX_MCO + 1` instead.

Reply:
yes, changing the size to `MAX_MCO + 1` is better, it would be strange to sometimes NUL-terminate the string but not always.